examples/roads2png.rs

```diff
@@ -73,66 +39,57 @@ impl Image {
     }
 }
 
-#[derive(Debug, Clone)]
-struct MapTransform {
-    min_x: f64,
-    max_y: f64,
-    scale_x: f64,
-    scale_y: f64,
+fn compute_bounds<'a>(mut iter: impl Iterator<Item = RefNode<'a>>) -> ((i64, i64), (i64, i64)) {
+    let first_coord = iter.next().map(|c| (c.lat(), c.lon())).unwrap_or((0, 0));
+    iter.fold((first_coord, first_coord), |(min, max), coord| {
+        (
+            (min.0.min(coord.lat()), min.1.min(coord.lon())),
+            (max.0.max(coord.lat()), max.1.max(coord.lon())),
+        )
+    })
 }
 
-impl MapTransform {
-    fn new(width: u32, height: u32, min: GeoCoord, max: GeoCoord) -> Self {
-        Self {
-            min_x: min.lon,
-            max_y: max.lat,
-            scale_x: (max.lon - min.lon) / (width as f64),
-            scale_y: (max.lat - min.lat) / (height as f64),
-        }
-    }
-
-    fn transform(&self, coord: GeoCoord) -> (isize, isize) {
+fn map_transform<'a>(
+    width: u32,
+    height: u32,
+    min: (i64, i64),
+    max: (i64, i64),
+) -> impl FnMut(RefNode<'a>) -> (isize, isize) + Copy {
```

Comment:
I have to say, here I also rather prefer a type which seems to be much cleaner to me. Returning lambdas with impl Trait looks slightly obfuscating to me, especially in an example.

Reply:
Actually I feel the other way round: You have a rather large type to do a 2-line conversion

Reply:
Ok, no strong opinion, so leave like it is.